app/models/asset.rb

```diff
@@ -154,12 +155,23 @@ def size_from_file
     file_stat.size
   end
 
-  def update_indirect_replacements
-    return if replacement.blank?
+  def update_indirect_replacements_on_publish
+    return unless changes[:draft] && !draft?
 
-    Asset.where(replacement_id: self.id).each do |asset|
-      asset.replacement = replacement
-      asset.save
+    Asset.where(replacement_id: self.id).each do |replaced_by_me|
+      Asset.where(replacement_id: replaced_by_me.id).each do |indirectly_replaced_by_me|
+        indirectly_replaced_by_me.replacement = self
+        indirectly_replaced_by_me.save
+      end
+    end
+  end
+
+  def backpropagate_replacement
+    return if replacement.blank? || replacement.draft?
+
+    Asset.where(replacement_id: self.id).each do |replaced_by_me|
+      replaced_by_me.replacement = self
+      replaced_by_me.save
     end
```

Comment:
I'm not sure this here changes anything... for assets where they're replaced by this asset, it sets them to be replaced by this asset?

Reply:
Whoops, that should be `replaced_by_me.replacement = self.replacement`

Reply:
Ah, cool :+1: